GitUp/Application/Document.m

```diff
@@ -1279,6 +1279,8 @@ - (BOOL)windowController:(GIWindowController*)controller handleKeyDown:(NSEvent*
           handled = YES;
         } else if (_snapshotsView.superview) {
           [self toggleSnapshots:nil];
+          // Correct button toggle state if toggled by esc
+          [(NSButton *)_snapshotsItem.view setState:NSOffState];
```

Comment:
Thanks for fixing this! Could I ask that we either do an if conditional on the `_snapshotsItem.view` to `NSButton` conversion or perhaps better yet we make it a computed property somewhere called `snapshotButton` that we safely cast to `NSButton`. I just prefer not to crash in a case where this view was later wrapped or changed in the future.

Reply:
You were more than right, as this would not work if f.e. later a shortcut would be used to open/close tha shortcuts view. I now safely cast (and inform future devs if the layout ever changes) and just synchronize the state of the view with the model so it will display correct state regardles of what was used to open/cole the pane.
I also added a tooltip as I noticed it was missing.